src/reporters/apiary-reporter.coffee

```diff
@@ -18,35 +19,41 @@ class ApiaryReporter
     @startedAt = null
     @endedAt = null
     @remoteId = null
+    @config = config
     @reportUrl = null
     @configureEmitter emitter
-    @inlineErrors = inlineErrors
     @errors = []
-    @verbose = process.env['DREDD_REST_DEBUG']?
+    @verbose = @env('DREDD_REST_DEBUG')?
     @configuration =
-      apiUrl: process.env['APIARY_API_URL'] || "https://api.apiary.io"
-      apiToken: process.env['APIARY_API_KEY'] || null
-      apiSuite: process.env['APIARY_API_NAME'] || 'public'
+      apiUrl: @env('APIARY_API_URL') || "https://api.apiary.io"
+      apiToken: @env('APIARY_API_KEY') || null
+      apiSuite: @env('APIARY_API_NAME') || 'public'
 
     logger.info 'Using apiary reporter.'
 
+  env: (property) ->
+    if not property?
+      return Object.keys(@config?.options?.custom?.env or process.env)
```

Comment:
Does this mean that JS API of Dredd must be initiated with the config for custom configuration something like:
```
config = 
  options:
    custom:
      env: 
        'APIARY_API_KEY': "secretkey3456789"
```

I don't like that encapsulation, under `env` key.